src/components/pages/users.js

```diff
@@ -3,38 +3,59 @@ import firstBy from 'thenby';
 import Page from '../page.js';
 import autoBind from 'auto-bind';
 import PropTypes from 'prop-types';
-import { Table, Popover } from 'antd';
-import { ROLE } from '../../domain/role.js';
 import Logger from '../../common/logger.js';
+import { ROLE } from '../../domain/role.js';
+import { Table, Popover, Menu } from 'antd';
 import { withTranslation } from 'react-i18next';
 import { inject } from '../container-context.js';
 import errorHelper from '../../ui/error-helper.js';
 import UserRoleTagEditor from '../user-role-tag-editor.js';
 import UserApiClient from '../../services/user-api-client.js';
-import { SaveOutlined, CloseOutlined } from '@ant-design/icons';
 import CountryFlagAndName from '../localization/country-flag-and-name.js';
 import UserLockedOutStateEditor from '../user-locked-out-state-editor.js';
 import { userShape, translationProps } from '../../ui/default-prop-types.js';
 
 const logger = new Logger(import.meta.url);
 
-const availableRoles = [
-  { name: ROLE.user, isReadonly: false },
-  { name: ROLE.admin, isReadonly: true }
-];
+const MenuItem = Menu.Item;
+
+const availableRoles = Object.values(ROLE);
+
+const TABS = {
+  internalUsers: 'internal-users',
+  externalUsers: 'external-users'
+};
+
+function separateInternalAndExternalUsers(allUsers) {
```

Comment:
You usually separate "something from something" :).

Reply:
That would sound to me more like extracting, when you separate A from B you keep/favor A.
Separate A and B sounds correct to me, you put in a list and you return it as split/separated in 2.

Reply:
I renamed it to `split...`, maybe then it's clearer that both subsets are equally important